nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateFlowFile.java

```diff
@@ -111,8 +116,19 @@ public class GenerateFlowFile extends AbstractProcessor {
                     + "per batch of generated FlowFiles")
             .required(false)
             .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .dependsOn(DATA_FORMAT, DATA_FORMAT_TEXT)
+            .dependsOn(UNIQUE_FLOWFILES, "false")
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
             .build();
+    public static final PropertyDescriptor CUSTOM_CONTENT = new PropertyDescriptor.Builder()
+            .name("Custom Content")
+            .description("Path to a file. If specified, this file will be used as content of the generated FlowFiles and the File Size will be ignored.")
+            .required(false)
+            .identifiesExternalResource(ResourceCardinality.SINGLE, ResourceType.FILE, ResourceType.URL, ResourceType.TEXT)
```

Comment:
Doesn't this warrant declaring the processor to require `READ_FILESYSTEM` permission? Is that a problem in regards to backwards compatibility?

Reply:
Looks like @exceptionfactory beat me to it.